didcomm_messaging/resolver/jwk.py

```diff
@@ -0,0 +1,59 @@
+"""did:jwk Resolver."""
+
+import re
+import json
+
+from didcomm_messaging import DIDResolver
+from didcomm_messaging.resolver import DIDResolutionError
+from didcomm_messaging.multiformats.multibase import Base64UrlEncoder
+
+b64 = Base64UrlEncoder()
+
+
+class JWKResolver(DIDResolver):
+    """Resolve did:jwk."""
+
+    PATTERN = re.compile(r"^did:jwk:(?P<did>.*)$")
+
+    async def resolve(self, did: str) -> dict:
+        """Resolve a did:jwk."""
+        if match := self.PATTERN.match(did):
+            encoded = match.group("did")
+        else:
+            raise DIDResolutionError(f"Invalid DID: {did}")
+
+        jwk = json.loads(b64.decode(encoded))
```

Comment:
Since the pattern being matched against is merely `r"^did:jwk:(?P<did>.*)$"`, I believe this allows for 0 or more characters of any type. Therefore, would it be better to have the JWK decode and parse line to happen in a try-catch, similar to this if-else statement?